pkg/exporter/sumologicexporter/sender.go

```diff
@@ -164,47 +164,61 @@ func (s *sender) send(ctx context.Context, pipeline PipelineType, body io.Reader
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
-		switch resp.StatusCode {
-		case 401:
-			return errUnauthorized
-		default:
-			return fmt.Errorf("error during sending data: %s", resp.Status)
-		}
+	return s.handleReceiverResponse(resp)
+}
+
+func (s *sender) handleReceiverResponse(resp *http.Response) error {
+	if resp.StatusCode >= 200 && resp.StatusCode < 400 {
+		return nil
 	}
 
-	s.handleReceiverResponse(resp)
+	switch resp.StatusCode {
+	case 401:
+		return errUnauthorized
+	default:
+		type ReceiverResponse struct {
+			Status  int    `json:"status,omitempty"`
+			ID      string `json:"id,omitempty"`
+			Code    string `json:"code,omitempty"`
+			Message string `json:"message,omitempty"`
+			Errors  []struct {
+				Code    string `json:"code"`
+				Message string `json:"message"`
+			} `json:"errors,omitempty"`
+		}
```

Comment:
Do you know if these fields are actually allowed to be empty? We could make the parsing and error-generation logic simpler if we knew they're always available.

Reply:
They are not. It depends on the error being returned.